### PR TITLE
(accounts) Do not allow at symbol in username

### PR DESCRIPTION
**Pull request by sastels**
https://www.notion.so/astrobindotcom/Disallow-sign-in-new-usernames-fc1f3b5c0061406b881b5965a817ee92?pvs=4

Don't allow "@" in username

Now get error:

image

### CODE REVIEW

File: astrobin/views/registration.py [hunk 1]
```diff
@@ -136,6 +136,10 @@ def clean_username(self):
             raise forms.ValidationError(
                 _('Sorry, this username already exists with a different capitalization.')
             )
+        elif "@" in value:
+            raise forms.ValidationError(
+                _('Sorry, your username cannot contain the "@" character.')
```
Comment (sastels) [inline astrobin/views/registration.py:141]: Probably have to add this string to the i18n files? is there a tool for this?

Comment (siovene) [reply astrobin/views/registration.py:141]: You don't need to, I generate them periodically and it happens outside of the CI.

**Review by siovene — CHANGES_REQUESTED** [review]

File: astrobin/views/registration.py [hunk 2]
```diff
@@ -136,6 +136,10 @@ def clean_username(self):
             raise forms.ValidationError(
                 _('Sorry, this username already exists with a different capitalization.')
             )
+        elif "@" in value:
```
Comment (siovene) [inline astrobin/views/registration.py:139]: You could get here if `value is None` and raise a `TypeError`. I think the cleanest way would be to check if value is None at the binning of the function, separately.

**Review by siovene — CHANGES_REQUESTED** [review]

File: astrobin/views/registration.py (hunk 2 above)
Comment (sastels) [reply astrobin/views/registration.py:137]: oh, I see, there's a silent fall-through to `return None` in the original code if value is None. 👍

**Comment by sastels** [comment]
Error cases now:
username is empty:
image

username already used:
image

username contains an "@":
image

**Review by siovene — APPROVED** [review]
Awesome, thnaks!

**Pull request merged**